index.js

```diff
@@ -2,14 +2,60 @@ const Vis = require("./graphvis");
 const { Module, render } = require("./visRenderer");
 
 let Sequelize;
-const relationship = ({ source, target, associationType, as }) => {
-  const typeString = {
-    BelongsTo: `[arrowtail=crow, arrowhead=none, dir=both, arrowsize=0.60]`,
-    BelongsToMany: `[arrowtail=crow, arrowhead=crow, dir=both, arrowsize=0.60]`
-  }[associationType];
-  if (typeString) {
-    return `"${source.name}" -> "${target.name}" ${typeString}`;
-  }
+
+const relationships = associations => {
+  const mappings = [];
+
+  const mapper = {
+    associationTypes: new Map([
+      ['BelongsToMany', { typeArrowShapes: ['none', 'crow'], items: [] }],
+      ['BelongsTo', { typeArrowShapes: ['crow', 'none'], items: [] }],
+      ['HasMany', { typeArrowShapes: ['none', 'crow'], items: [] }],
+      ['HasOne', { typeArrowShapes: ['none', 'none'], items: [] }],
+    ]),
+  };
+
+  // Separation of association by type to process in the right order
```

Comment:
I'm super close to understanding why this is needed, but i'm not quite there. Could you give an example here as a comment?

Reply:
I'm not sure if I can provide an example, but I think I figured this section out. After trying out a few things, it looks like this approach here is a clever way to remove duplicated lines that were previously removed in the `.filter()` part of `generateDot()`'s return value—it populates an array (`mappings`) with new values instead of filtering an existing array down. Since it moves from a series of mappings to a single function processing all `associations`, it also changes how the relationships are processed. It also processes the relationships in a specific order from most to least interconnectedness, which makes sense since the `mappings` array is getting populated as it goes through each association.

There are a couple of things I would have done differently for clarity, for example, `thisMapping` uses a `filter()` where I would have done a `find()`, but it works the same and arrays might be marginally faster to work with, so I don't think it's an issue.

I feel like the deduplication process could be rewritten to be a little bit more clear, but I'm not immediately sure how that might be done without making it less concise. This is very concise and clean, but it takes a while to read through everything to figure it out!

Either way, it all works very well and adds in the `HasMany` and `HasOne` cases that I use extensively in my model associaitons, so I would recommend accepting it. :)